src/main/java/de/embl/cba/mobie/n5/S3Reader.java

```diff
@@ -0,0 +1,51 @@
+package de.embl.cba.mobie.n5;
+
+import de.embl.cba.mobie.n5.open_organelle.OpenOrganelleN5S3ImageLoader;
+import de.embl.cba.mobie.n5.open_organelle.OpenOrganelleS3Reader;
+import ij.IJ;
+import mpicbg.spim.data.SpimData;
+import net.imglib2.util.Cast;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public abstract class S3Reader {
```

Comment:
In general we should try to avoide code duplication and introduce more base classes for the N5 logic.
However, you introduce a dependency on the open organelle loaders here, which is not a good idea.
I would suggest you first implement all logic in the `OpenOrganelleS3Reader` and we think about how to avoid code duplication (together with @tischi) in a later PR.

Reply:
Or maybe we can just use "S3ImageLoader" here and don't need to reimplement it in open organelle or ome zarr.